### ISSUE TITLE
🐛 BUG: `@astrojs/compiler` panic

### ISSUE THREAD

**Issue by MoustaphaDev** [withastro/astro#5003]
### What version of `astro` are you using?

1.4.4

### Are you using an SSR adapter? If so, which one?

None

### What package manager are you using?

npm

### What operating system are you using?

Ubuntu 22.04

### Describe the Bug

`@astrojs/compiler` encountered an unrecoverable error when compiling the following file.

**src/components/SillyTable.astro**
```astro
---
const  MyTableRow = "tr";
---

<table>
    <MyTableRow>
        <td>Witch</td>
    </MyTableRow>
    <MyTableRow>
        <td>Moon</td>
    </MyTableRow>
</table>
```

Obviously the markup isn't correct, this the minimal code in which I could get same the compiler error. When I remove the `table` tag, I don't get the compiler error anymore. I guess the compiler doesn't expect components as children of a table?

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-yv5nwx?file=src/pages/index.astro

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by matthewp** [comment]
Yeah I think this is probably just not allowed by our compiler, but I don't think it should cause a panic. cc @natemoo-re
